src/Bounds.cpp

```diff
@@ -1530,29 +1530,47 @@ Box box_touched(Stmt s, string fn, const Scope<Interval> &scope, const FuncValue
     return box_touched(Expr(), s, true, true, fn, scope, fb);
 }
 
+// Compute interval of all possible function's values (default + specialized values)
+Interval compute_pure_function_definition_value_bounds(
+        const Definition &def, const Scope<Interval>& scope, const FuncValueBounds &fb, int dim) {
+
+    Interval result = bounds_of_expr_in_scope(def.values()[dim], scope, fb);
+
+    // Pure function might have different values due to specialization.
+    // We need to take the min and max bounds of all those possible values.
+    for (const auto &s : def.specializations()) {
+        Interval s_interval = compute_pure_function_definition_value_bounds(s.definition, scope, fb, dim);
+        if (result.min.defined()) {
```

Comment:
There's *probably* a helper function that does this combination of two intervals, it must be a very common operation in bounds inference.